react/fields/TextField/TextField.js

```diff
@@ -163,64 +167,58 @@ export default class TextField extends Component {
     );
   }
 
-  renderHelp() {
-    const { message, help } = this.props;
-
-    if (message || !help) {
-      return;
-    }
-
-    const { helpProps } = this.props;
-    const allHelpProps = combineClassNames(helpProps, styles.help);
-
-    return (
-      <p {...allHelpProps}>
-        {help}
-      </p>
-    );
-  }
-
   renderMessage() {
-    const { message } = this.props;
+    const { message, valid } = this.props;
 
     if (!message) {
       return;
     }
 
-    const { messageProps } = this.props;
-    const allMessageProps = combineClassNames(messageProps, styles.message);
+    const { critical, positive, secondary, ...restMessageProps } = this.props.messageProps;
+    const allMessageProps = combineClassNames(restMessageProps, styles.message);
 
     return (
-      <p {...allMessageProps}>
-        {this.renderIcon()}
+      <Text {...allMessageProps} critical={(valid === false && !secondary) || critical} positive={(valid === true && !secondary) || positive} secondary={typeof valid === 'undefined' || secondary}>
+        {this.renderMessageIcon()}
         {message}
-      </p>
+      </Text>
     );
   }
 
-  renderIcon() {
-    const { invalid } = this.props;
+  renderMessageIcon() {
+    const { valid } = this.props;
+
+    if (valid === false) {
+      return (
+        <ErrorIcon
+          filled={true}
+          className={styles.messageIcon}
+          svgClassName={styles.messageIconSvg}
+        />
+      );
+    }
 
-    if (!invalid) {
-      return;
+    if (valid === true) {
+      return (
+        <TickCircleIcon
+          filled={true}
+          className={styles.messageIcon}
+          svgClassName={styles.messageIconSvg}
+        />
+      );
     }
 
-    return (
-      <ErrorIcon
-        filled={true}
-        className={styles.messageIcon}
-        svgClassName={styles.messageIconSvg}
-      />
-    );
+    return;
```

Comment:
I suggested `return null;` given this function is being called from a template, and there is a possibility of not entering either of the if statements.